### ISSUE TITLE
Inline code in text cell

### ISSUE THREAD

**Issue by LeonardoGentile** [HappenApps/Quiver#19]
Hello again, I have another suggestion :)
What about inline code in text cell? 

The only problem would be how to identify it..maybe with the `backtick` character  as we alredy use here in github issues. The difference is that here we are using some sort of markdown and in a text cell..well only text. 

But I guess that after all wouldn't be so difficult to scan text for `` `example` `` and render inline code that it identifies

**Comment by ylian** [comment]
A text cell is WYSIWYG, so `backticks` don't belong there. GitHub converts backticks to inline code because it passes the text through a Markdown compiler. Quiver does that for Markdown cells.

I also want to avoid scanning text as much as possible. It slows down the editing experience. Little things like this do add up.

But one idea might work: a formatting option for "inline code", just like "bold", "italics", with a keyboard shortcut. Might be possible.

**Comment by LeonardoGentile** [comment]
Yes you are right about scanning the text..
The formatting option (especially with a shortcut) would be very useful! :+1:

**Comment by michael-jarosik** [comment]
Agreed - a formatting option for inline code sounds like a great idea

**Comment by ylian** [comment]
Tried but it's quite tricky to cover all the edge cases. Shelved for now.

**Comment by LeonardoGentile** [comment]
Any news on this?

**Comment by ylian** [comment]
Still shelved. Will try again.

**Comment by iby** [comment]
:+1: Would be nice to have. I'm trying to stick to a single workflow and avoid using markdown when not needed, but inline code seems to be only possible in there.

**Comment by robdecker** [comment]
I found a little trick. Convert the cell to markdown, then `backtick` what you want, then convert back to a text cell. Code formatting remains.

**Comment by LeonardoGentile** [comment]
Any news on this?

**Comment by ylian** [comment]
Yes, implemented already. I will announce public beta of Quiver 3.0 later today.

**Comment by LeonardoGentile** [comment]
:+1: Awesome!!

**Comment by preflight** [comment]
Yeah that's great news. Really looking forward to 3.0

**Comment by ylian** [comment]
Quiver 3.0 beta: https://github.com/HappenApps/Quiver/issues/162

**Comment by LeonardoGentile** [comment]
A remark: it's not easy to remove an inline code. I mean after being created, if the cursor is within the inline code, at any point within it and I hit `ctr-shift-k` I would have expected that the inline code transformed back to plain text, but this is not the case. To do so I have to select (being careful not to miss or add any space) the whole inline code and then hit the shortcut.

Also, is there any way to customize the aspect of the inline code? Like deciding what theme to apply to it? Or at least chose between dark and light? Or editing its css anywhere?

Thanks

**Comment by ylian** [comment]
That's a bug. I will fix right away. Please report any issue like this. I built this one from scratch, so there may still be things I haven't considered. (If the inline code is highlighted, pressing `ctr-shift-k` does work.)

You can customize the appearance like this:

For dark theme:

``` css
.dark code {
  color: red;
  background-color: blue;
  border: 1px solid green;
  padding: 2px 5px;
}
```

Use `.light code` for light theme, or `code` for both.

Put the CSS in "Preferences > Styles > Editor CSS".

If you find a nice style, please share here so I might change the default styles.

**Comment by ylian** [comment]
Might have to take this feature out from 3.0. There are other more serious issues: detecting inline code style throws a JavaScript error in some HTML.

**Comment by ylian** [comment]
I figured out the JavaScript error so this is still in.

@LeonardoGentile Regarding the behavior of hitting `ctrl-shift-k` while the cursor is inside the inline code (but no selection), it is actually consistent with other formatting controls such as bold, italic.

**Comment by LeonardoGentile** [comment]
@ylian (IMO) It doesn't feel very smooth to make a perfect selection to remove inline text 

There is a little bug tough:
- create an inline code
- copy and paste it anywhere else
- remove the inline code 

Result:
when you try to remove the inline code from the pasted one (ctrl-shift-k) it doens't remove it. If you write it is still an inline code but the font size is smaller. If you trye to remove it again it goes back to normal font size but still inline code and so on. 
A couple of time I've managed to reproduce this but applying the same process as described above I've obtained a series of inline code cells that became smaller and smaller in term of font size

![quiver](https://cloud.githubusercontent.com/assets/412061/10215593/32b9bdde-6823-11e5-9d2b-176a6f7bd2d0.png)

**Comment by peterkrieg** [comment]
@LeonardoGentile @ylian  any way to currently do this?  I know a text cell is "WYSIWYG" but coders are so used to using backticks in text.. whether it be here on github, slack, jira, etc.. I think it would be a really popular feature.  Most people never need to use backticks in text anyways.  It could be toggleable like how vscode lets you turn off all new features.

**Comment by LeonardoGentile** [comment]
@peterkrieg that would awesome also for me.
Unfortunately @ylian has gone silent for a long time, I'm not sure if this app is still supported (683 open issues)

**Comment by koskokos2** [comment]
> has gone silent for a long time, I'm not sure if this app is still supported

He is just busy finishing new version. It is possible to get support via email with a prompt answer from him. I think he will review issue after new version will be released as it involves major refactoring and issues will need to be revalidated afterwards.

**Comment by peterkrieg** [comment]
Maybe we could have a system where issues that are most popular (by voting, like 👍  how other projects do it) could be tackled first.  

Hopefully this will still be maintained, I really love the app so far!

Otherwise seems like an open-source alternative might be the way to go.

**Comment by haichao-yu** [comment]
@ylian Is there a way to change the style of inline code in markdown? For example, change the color of the inline code to be red. Thanks!
